doc/design/draft-metadata-migration.md

```diff
@@ -0,0 +1,15 @@
+Premise: Draft metadata should always be in the latest schema version.
+
+There are a few places in the API where draft metadata are altered (POST, PUT) and served (GET). Note that the UI and 
+API use a shared schema version. All actions should thus assume that this is the only version available to the UI and
+server. Any other versions need to be migrated to this version for any display or editing.
+
+This proposal only applies to the `schemaVersion` of dandiset metadata, not asset metadata.
+
+1. The API should call dandischema `migrate` on all metadata exchanges of draft metadata.
+2. The API can have the option (parameter `?migrate=true`) of providing migration on a `GET` on published metadata. 
```

Comment:
```suggestion
2. The API can have the option (parameter `?migrate=<version>|bool`) of providing migration on a `GET` on published metadata. 
```
and if `/migrate` then more consistent `to_version`: we should expose `to_version` kwarg of `dandischema.migrate` so a user can consistently migrate to desired (later) schema -- e.g. to establish a meta-analysis script (relying on specific schemaVersion) across some already published dandisets.

Reply:
I think that is a feature we can add later if there is demand for it.